apps/smart-contracts/token/test/MiniSales.test.ts

```diff
@@ -435,4 +455,180 @@ describe('=> MiniSales', () => {
+      await purchaseAllowlist.connect(owner).acceptOwnership()
+      await purchaseAllowlist
+        .connect(owner)
+        .set([owner.address, allowedPurchaser.address], [true, true])
+      // Setup allow list purchase hook
+      allowlistPurchaseHook = await allowlistPurchaseHookFixture(owner.address)
+      await allowlistPurchaseHook.connect(owner).acceptOwnership()
+      await allowlistPurchaseHook.connect(owner).setAllowlist(purchaseAllowlist.address)
+      // Setup Source allow list
+      sourceAllowlist = await accountListFixture(owner.address)
+      await sourceAllowlist.connect(owner).acceptOwnership()
+      await sourceAllowlist
+        .connect(owner)
+        .set(
+          [ZERO_ADDRESS, owner.address, miniSales.address, allowedSourceUser.address],
+          [true, true, true, true]
+        )
+      // Setup Destination allow list
+      destinationAllowlist = await accountListFixture(owner.address)
+      await destinationAllowlist.connect(owner).acceptOwnership()
+      await destinationAllowlist
+        .connect(owner)
+        .set([owner.address, allowedDestinationUser.address], [true, true])
+      // Setup Blocklist
+      blocklist = await accountListFixture(owner.address)
+      await blocklist.connect(owner).acceptOwnership()
+      await blocklist.connect(owner).set([blockedUser.address], [true])
+      // Setup RestrictedTransferHook
+      restrictedTransferHook = await restrictedTransferHookFixture(owner.address)
+      await restrictedTransferHook.connect(owner).acceptOwnership()
+      await restrictedTransferHook.connect(owner).setSourceAllowlist(sourceAllowlist.address)
+      await restrictedTransferHook
+        .connect(owner)
+        .setDestinationAllowlist(destinationAllowlist.address)
+      await restrictedTransferHook.connect(owner).setBlocklist(blocklist.address)
+      // Connect PPO with TransferHook
+      await ppo.connect(owner).setTransferHook(restrictedTransferHook.address)
+    }
+
+    beforeEach(async () => {
+      await setupEndToEndStack()
+      purchaser = user1
+      // Mint PPO to MiniSales
+      await ppo.connect(owner).mint(miniSales.address, parseUnits('100', saleTokenDecimals))
+      await miniSales.connect(owner).setPrice(testPrice)
+      await saleToken
+        .connect(owner)
+        .transfer(miniSales.address, parseUnits('100', saleTokenDecimals))
+      contractSaleTokenBalance = await saleToken.balanceOf(miniSales.address)
+      // To support sufficient balance for subsequent purchase
+      saleTokenAmount = contractSaleTokenBalance.sub(1)
+      paymentNeeded = saleTokenAmount.mul(testPrice).div(testDenominator)
+      await paymentToken.connect(owner).transfer(purchaser.address, paymentNeeded)
+      // As we give infinite approval to MiniSales
+      await paymentToken.connect(purchaser).approve(miniSales.address, MAX_UINT256)
+    })
+
+    it('reverts purchase if recipient blocked', async () => {
```

Comment:
```suggestion
    it('reverts purchase if recipient blocked by PPO', async () => {
```
After reading PR 302, I realize your organization scheme is essentially the same as previous PRs where you write the PPO access control conditions (i.e. recipient blocked), but later on will have `if sale permissioned...` nested, where MiniSales specific access control conditions are listed.

This is fine, but it can get confusing with different access controls, lets be specific in this test suite about which "contract stack"'s access controls we are referring to.